docs/Tutorial_DiscordMERLIN.ipynb

```diff
@@ -0,0 +1,1780 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #1.    def _find_candidates(T, m, M_T, Σ_T, min_dist, is_cands, direction):
</pre><p>Instead of <code>direction</code>, it would be simpler to use <code>right=True</code> and if <code>right=False</code> then this turns into <code>left</code> :</p><pre class="ql-syntax" spellcheck="false">def _find_candidates(T, m, M_T, Σ_T, min_dist, init_cands=None, right=False, finite=False)
</pre><p>This then allows us to avoid string comparisons within the code, avoids the dangling <code>if/elif</code>, and also simplifies it:</p><p> </p><pre class="ql-syntax" spellcheck="false">if right:
    non_trivial_cands_idx = cands_idx[cands_idx &lt; max(0, i - excl_zone)]
else:  # left
    non_trivial_cands_idx = cands_idx[cands_idx &gt; i + excl_zone]
</pre><p> </p><p>Unrelated to the <code>direction</code>, I am starting to change my mind about allowing these functions to return <code>np.inf</code> as the discord. In other areas of our code, it is possible to return <code>np.inf</code> as the discord and so we should be consistent and allow the same. I'm thinking that we should add a parameter called <code>finite=False</code> by default, which controls whether or not the discord must be finite:</p><p> </p><pre class="ql-syntax" spellcheck="false">T_subseq_isfinite = np.isfinite(M_T)
if not finite:  # allow np.inf discords
    T_subseq_isfinite[:] = True 
</pre><p> </p><p>Another unrelated point, perhaps we should call it <code>init_cands</code> in the function definition and then create <code>is_cands</code> inside of the function:</p><p> </p><pre class="ql-syntax" spellcheck="false">is_cands = np.ones(k, dtype=bool)
if init_cands is not None:
    is_cands[:] = init_cands 
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Reply:
> Instead of `direction`, it would be simpler to use `right=True` and if `right=False` then this turns into `left` :

Sure! I will change it.

Reply:
> I'm thinking that we should add a parameter called `finite=False` by default, which controls whether or not the discord must be finite

We can do it to make it consistent with stumpy output. But, I think there is one little (?) problem: `stumpy.stump` gives the matrix profile and thus we can find all infinite discords. However, in `_mutlin`, we return `top-k` discords. Let us assume we have 10 `inifinite` discords, how should we return top-5? I think we cannot simply compare them as all of them are `infinite`, and it may not make sense to return the left-most index first as the user might miss the others! So, I think we can just return all indices where `T_subseq_isfinite` is False. But, as we discussed, the user should be able to get this easily. 

Do you have any suggestion? How about adding a note in docstring to help user understand this difference?

Reply:
> we should call it `init_cands` in the function definition and then create `is_cands`

Yes! This is definitely more readable. Thanks for the suggestion! I will revise it.

Reply:
<blockquote>Let us assume we have 10&nbsp;<code>inifinite</code>&nbsp;discords, how should we return top-5? I think we cannot simply compare them as all of them are&nbsp;<code>infinite</code>, and it may not make sense to return the most-left index first as the user might miss the others!&nbsp;</blockquote><p>I think it is reasonable to return the 5 left-most <code>np.inf</code> values  </p>

Reply:
I will revise it.

Reply:
(Just for the record) I know we will review `_murlin(...)` later... just wanted to let you know that we can add parameter `finite` to the `_murlin` function (that returns `top-k` discords). So, we can first get `np.inf` discords (if requested by user) and then use `_find_candidates` to find finite candidates. 

(for now: I just added parameter `finite` to the function `_murlin(...)` as well. and I just pass it to `_find_candidates(...)` whenever it is called inside `_murlin(...)`)

Reply:
> Another unrelated point, perhaps we should call it `init_cands` in the function definition and then create `is_cands` inside of the function:

Are you ok with the name `is_cands_init`?

Reply:
<p>Cool. Thanks for the heads up</p>